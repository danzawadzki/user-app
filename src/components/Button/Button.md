React component example:

Default button.

```jsx
<Button id="test" onClick={() => console.log('Fired by onClick event')}>
	Click me
</Button>
```

Full width button.

```jsx
<Button
	id="test"
	isFullWidth={true}
	onClick={() => console.log('Fired by onClick event')}>
	Click me
</Button>
```
