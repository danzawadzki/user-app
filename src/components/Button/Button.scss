@import '../../assets/styles/colors.scss';
@import '../../assets/styles/minor.scss';

//TODO: Line overwrites rem to be multiple of 10px, not 16px. I keep it in the component styles, as I didn't know how to pass global styles to the react-styleguidist.
html {
	font-size: 62.5%;
}

.Button {
	padding: 1.4rem 2.7rem;
	border-radius: 10rem;
	background-color: $brand--accent;
	border: none;
	transition: $transition--default;
	font-family: 'Montserrat', sans-serif;
	font-weight: 600;
	font-size: 14px;
	color: #ffffff;
	letter-spacing: 4.2px;
	text-transform: uppercase;

	&:hover {
		filter: brightness(110%);
	}

	&:active {
		transition: $transition--default;
		transform: $transform__scale;
	}

	&--fullWidth {
		width: 100%;
	}
}
