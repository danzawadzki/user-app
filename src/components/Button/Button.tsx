import * as React from 'react';
import './Button.scss';

export interface IButton {
	/** DOM element id. */
	id: string;
	/** Flag to display fullwidth button. */
	isFullWidth?: Boolean;
	/** Button child html node. */
	children: String;
	/** Handler to the onClick event */
	onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

/**
 * Button component.
 *
 * @version 1.0.0
 */
const Button = ({ id, isFullWidth = false, children, onClick }: IButton) => (
	<button
		type="button"
		className={`Button ${isFullWidth && 'Button--fullWidth'}`}
		onClick={onClick}>
		{children}
	</button>
);

export default Button;
