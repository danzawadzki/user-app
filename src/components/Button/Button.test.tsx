import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button component: ', () => {
	//Shallow rendered component
	const label = 'Push me!';

	const component = shallow(
		<Button id="button_test" onClick={() => console.log('Fired by onClick event.')}>
			{label}
		</Button>
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly with all props', () => {
		expect(component.find('.Button')).toBeDefined();
	});

	it('should render with correct label', () => {
		expect(component.get(0).props.children).toEqual(label);
	});

	it('should render with full width styles', () => {
		const componentFullWidth = shallow(
			<Button
				id="button_test"
				isFullWidth={true}
				onClick={() => console.log('Fired by onClick event.')}>
				{label}
			</Button>
		);

		expect(componentFullWidth.find('.Button--fullWidth')).toBeDefined();
	});
});
