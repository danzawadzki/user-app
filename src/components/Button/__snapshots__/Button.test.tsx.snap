// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component:  should render match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    id="button_test"
    onClick={[Function]}
  >
    Push me!
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Push me!",
      "className": "Button false",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "Push me!",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Push me!",
        "className": "Button false",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "Push me!",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
